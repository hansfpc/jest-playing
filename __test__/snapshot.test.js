import { getCharacter } from '../snapshot'
import rick from '../rick'

describe('Probando con Snapshots!', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  })
  test('Siempre fallará el snapshot', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'Hans',
    }
    expect(user).toMatchSnapshot(
      /*Si no pongo este objeto, FALLARÁ */
      {
        id: expect.any(Number),
        createAt: expect.any(Date),
      }
    )
  })
  test('Tenemos una excepsion dentro del codigo', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Hans Felipe',
      createAt: new Date(),
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
      createAt: expect.any(Date),
    })
  })
})
