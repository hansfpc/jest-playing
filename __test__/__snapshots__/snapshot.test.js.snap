// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando con Snapshots! Siempre fallar√° el snapshot 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "Hans",
}
`;

exports[`Probando con Snapshots! Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": "1",
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Probando con Snapshots! Tenemos una excepsion dentro del codigo 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "Hans Felipe",
}
`;
